src/Access/User.cpp

```diff
@@ -14,4 +19,14 @@ bool User::equal(const IAccessEntity & other) const
         && (settings == other_user.settings) && (grantees == other_user.grantees) && (default_database == other_user.default_database);
 }
 
+void User::setName(const String & name_)
+{
+    /// Unfortunately, there is not way to distinguish USER_INTERSERVER_MARKER from actual username in native protocol,
+    /// so we have to ensure that no such user will appear.
+    /// Also it was possible to create a user with empty name for some reason.
+    if (name_ == USER_INTERSERVER_MARKER || name_.empty())
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "User name is reserved");
```

Comment:
It's better to write the name in the message too.
And it seems an empty name is not reserved, it's just not valid.